### ISSUE TITLE
[Bug]: "sshd requires execution with an absolute path"

### ISSUE THREAD

**Issue by martin-braun** [termux/termux-packages#20756]
### Problem description

Since the last upgrade my sshd won't start anymore. It outputs "sshd requires execution with an absolute path". Searching online revealed that this is a security measure against attackers who like to inject sshd in malicious scripts and that the solution is to use a proper service runner such as `systemd`. (I had sshd in my `.profile` and called it a day.)

### What steps will reproduce the bug?

I then installed [termux-services](https://wiki.termux.com/wiki/Termux-services), restarted Termux and enabled sshd via `sv-enable sshd`, but sadly the issue is not solved. Now my `$PREFIX/var/log/sv/sshd/current` is cluttered with the same error message. The `runit` scripts seem not to solve this issue.

### What is the expected behavior?

`sshd` should launch successfully.

### System information

```shell
Termux Variables:
TERMUX_APP_PACKAGE_MANAGER=apt
TERMUX_APP_APK_FILE=/data/app/~~FMnKSX-RECIuHgYJbY1kjA==/com. termux-uqn4WjXy•gORj
vla2Pdwog==/base.apk
TERMUX_APP__APK_RELEASE=F_DROID
TERMUX_APP__APR_VERSION_CODE=1020
TERMUX_APP__APP_VERSION_NAME=0.119.0-beta.1
TERMUX_APP__DATA_DIR=/data/user/0/com.termux
TERMUX_APP__IS_DEBUGGABLE_BUILD=false
TERMUX_APP__IS_INSTALLED_ON_EXTERNAL_STORAGE=false
TERMUX_APP__PACKAGE_NAME=com.termux
TERMUX_APP__PID=4261
TERMUX_APP__TARGET_SDK=28
TERMUX_VERSION=0.119.0-beta. 1
TERMUX__SE_FILE_CONTEXT=u:object_r:app_data_file:s0:c189,c256,c512,c768
TERMUX__SE_INFO=default:targetSdkVersion=28:complete
TERMUX__SE_PROCESS_CONTEXT=u:r:untrusted
TERMUX__UID=10189_app_27:50:0189, 0256, 0512, C/68
TERMUX__USER_ID=0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources. list
deb https://mirror.sunred.org/termux/termux-main stable main
Updatable packages:
All packages up to date termux-tools version:
1.42.4
Android version:
13
Kernel build information:
Linux localhost 4.4.302-dos.p383+ #1 SMP PREEMPT Sat Jun 15 04:07:31 EDT 2024 aarc
h64 Android
Device manufacturer:
OnePlus
Device model:
ONEPLUS A5000
LD Variables:
LD_LIBRARY_PATH=
LD_PRELOAD=/data/data/com.termux/files/usr/lib/libtermux-exec.so
Installed termux plugins:
com.termux.boot versionCode: 1000
```

**Comment by 2096779623** [comment]
![image](https://github.com/termux/termux-packages/assets/57583560/413f5578-bd4a-442d-8d0a-136825806185)
It worked now.

**Comment by 2096779623** [comment]
> @2096779623 Password authentication doesn't work anymore.
> 
> Compared ssh from [2f33196](https://github.com/termux/termux-packages/commit/2f33196a0b2d30eb4eb98a86847ffde2a25a0721) and my backup.

I also found this problem and was looking for the cause.

**Comment by martin-braun** [comment]
@2096779623 Thank for the quick action from you. Yes, the recent update fixed the error message, but `sshd` is not working for me for some reason. 

I used to run `sshd` to then connect via USB wired connection to it using this command:

```sh
adb forward tcp:8022 tcp:8022
ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o GlobalKnownHostsFile=/dev/null localhost -p 8022
```

The forwarding works as expected, but the `ssh` command hangs indefinitely. `$HOME/.ssh/authorized_keys` on my device is still in place and wasn't changed.

From Termux itself, I can also not ssh onto itself using `ssh localhost -p 8022` with the same behavior. It hangs indefinitely.

**Comment by martin-braun** [comment]
@sylirre I don't like what has happened, but it magically resolved itself. I recalled, that I didn't change anything and the OpenSSH upgrade that caused me to open that issue was the event that caused things to break, so much so that the port was blocked for no known reason. No apps were installed or uninstalled. My device was restarted several times, nothing worked.

Today I restarted my device once again and I tested to set `Port 8023` in my `sshd_config` and tada, it's working. Going back to `Port 8022` to tackle the blocked port issue ... and nothing seems to block port 8022 anymore.

I have no idea what the culprit to all of this was, and that is what I don't like about. I haven't changed the system, and I did try to restart the device earlier without any success. I don't know if changing the port temporary fixed the problem or the restart itself and it gives me chills to think about it, because it gives me reason to believe that there is still any bug that caused the port to be blocked for no reason. 

I definitely do NOT have any other app that is blocking any port. This is a very simple device with very little apps. Most off the stuff is even locked in shelter. We talking about a privacy respecting DivestOS phone with a minimalistic approach. It's rather a toolbox with phone capabilities and less of a regular smart phone.
